sympy/core/expr.py

```diff
@@ -1956,6 +1956,11 @@ def extract_multiplicatively(self, c):
             x = self.extract_multiplicatively(a)
             if x is not None:
                 return x.extract_multiplicatively(b)
+
+        from sympy.core.function import _coeff_isneg as f
```

Comment:
You can just import this; don't assign to `f`. I just did that because it looked cleaner in the snippet that I sent (and that's how I tested it before I sent it to the issue page). Try to put the import at the beginning of the file (if it's not already there).